app/javascript/components/import-image-form.jsx

```diff
@@ -0,0 +1,100 @@
+          var provider_classes = supported_providers;
+
+          API.get('/api/providers?expand=resources&attributes=id,name,type').then(({resources}) => {
+              resources.forEach((provider) => {
+                  if(provider['id'] === ManageIQ.record.recordId) return;
+
+                  var result = provider_classes.find(provider_class => provider_class['type'] === provider['type']);
+
+                  if (typeof result !== typeof undefined && result['kind'] === kind)
+                      options.push({value: provider['id'], label: provider['name']});
+              })
+
+              resolve(options);
+          })
+      })
+  })
+}
+
+const fetchImages = (provider) => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_templates?expand=resources&attributes=id,name&filter[]=ems_id=' + provider).then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const fetchDiskTypes = () => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_volume_types?expand=resources&attributes=id,name').then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const fetchBuckets = (provider) => {
+    return new Promise((resolve, reject) => {
+        API.get('/api/cloud_object_store_containers?expand=resources&attributes=name,ems_id&filter[]=ems_id=' + provider).then(({resources}) => {
+            let options = resources.map(({id, name}) => ({value: id, label: name}));
+            options.unshift(BLANK);
+            resolve(options);
+        })
+    })
+}
+
+const ImportImageForm = ({ dispatch }) => {
+  const [provider, setProvider] = useState('-1');
+  const [storage,  setStorage]  = useState('-1');
+  const [image,    setImage]    = useState('-1');
+  const [bucket,   setBucket]   = useState('-1');
+  const [diskType, setDiskType] = useState('-1');
+  const [keepOva,  setKeepOva]  = useState(false);
+
+  const providers = fetchProviders('cloud');
+  const storages  = fetchProviders('storage');
+  const diskTypes = fetchDiskTypes();
+  const images    = useMemo(() => fetchImages(provider), [provider]);
+  const buckets   = useMemo(() => fetchBuckets(storage), [storage]);
+
+  const initialize = (formOptions) => {
+    dispatch({ type: "FormButtons.init",        payload: { newRecord: true, pristine: true } });
+    dispatch({ type: "FormButtons.customLabel", payload: __('Import') });
+    dispatch({ type: 'FormButtons.callbacks',   payload: { addClicked: () => formOptions.submit() }});
+  };
+
+  const submitValues = () => {
+      if ([provider, image, storage, bucket].some(item => parseInt(item) === -1))
+      {
+          window.add_flash("Request ignored due to incomplete input", 'warning');
```

Comment:
You can add      ` isRequired: true,validate: [{ type: validatorTypes.REQUIRED }],` to schema , ifuser did not enter these fields, this will automatically disable submit button .  I don't we need to a flash message here.

Reply:
I have removed this code and use my own custom validator as the standard "required" one is not sufficient in my case. Now the "Submit" button is enabled only if the required fields pass the validation test and this code above is removed.